dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/docker/BoulderContainer.java

```diff
@@ -0,0 +1,253 @@
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.MountableFile;
+
+import com.github.dockerjava.api.model.ContainerNetwork;
+import com.github.dockerjava.api.model.ContainerNetwork.Ipam;
+import com.ibm.websphere.simplicity.log.Log;
+import com.ibm.ws.security.acme.utils.AcmeFatUtils;
+
+public class BoulderContainer extends CAContainer {
+
+	public static final int HTTP_PORT = 5002;
+
+	/** The port used to listen for incoming ACME requests. */
+	public static final int LISTEN_PORT = 4001;
+
+	/** The REST management API port. */
+	public static final int MANAGEMENT_PORT = 8055;
+    
+    public final Network bluenet = Network.builder()
+            .createNetworkCmdModifier(cmd -> {
+                cmd.withDriver("bridge");
+                cmd.withIpam(new com.github.dockerjava.api.model.Network.Ipam()
+                        .withDriver("default")
+                        .withConfig(new com.github.dockerjava.api.model.Network.Ipam.Config()
+                                .withSubnet("10.77.77.0/24")));
+            })
+            .build();
+
+    public final Network rednet =  Network.builder()
+            .createNetworkCmdModifier(cmd -> {
+                cmd.withDriver("bridge");
+                cmd.withIpam(new com.github.dockerjava.api.model.Network.Ipam()
+                        .withDriver("default")
+                        .withConfig(new com.github.dockerjava.api.model.Network.Ipam.Config()
+                                .withSubnet("10.88.88.0/24")));
+            })
+            .build();
+    
+    public final GenericContainer<?> bhsm = new GenericContainer<>(IMAGE)
+            .withEnv("PKCS11_DAEMON_SOCKET", "tcp://0.0.0.0:5657")
+            .withExposedPorts(5657)
+            .withNetwork(bluenet)
+            .withNetworkAliases("boulder-hsm")
+            .withCommand("/usr/local/bin/pkcs11-daemon /usr/lib/softhsm/libsofthsm2.so")
+            .withLogConsumer(o -> System.out.print("[HSM] " + o.getUtf8String()));
+    
+    public final GenericContainer<?> bmysql = new GenericContainer<>("mariadb:10.3")
+            .withNetwork(bluenet)
+            .withExposedPorts(3306)
+            .withNetworkAliases("boulder-mysql")
+            .withEnv("MYSQL_ALLOW_EMPTY_PASSWORD", "yes")
+            .withCommand("mysqld --bind-address=0.0.0.0 --slow-query-log --log-output=TABLE --log-queries-not-using-indexes=ON")
+            .withLogConsumer(o -> System.out.print("[SQL] " + o.getUtf8String()));
+    
+    private static final String IMAGE = "ryanesch/acme-boulder:latest";
```

Comment:
Rename to DOCKER image? May want to comment that you create this one as part of replicating docker-compose with a GenericContainer.

I wonder if WebSphere has it's own docker hub we could use?

Reply:
We also probably need instruction on how you created your image.